### PR TITLE
Update Haskell to LTS 22.9

### PR DESCRIPTION
**Pull request by td202**
## Purpose

Update to GHC 9.6.4, LTS 22.9.

## Changes

- Update concordium-client
- Clean up dependencies
- Update `http2-client` and `http2-grpc-haskell`
- Fix proto-lens related dependencies (outside current stackage).

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [X] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [X] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: package.yaml [hunk 1]
```diff
@@ -88,8 +77,10 @@ executables:
     - -with-rtsopts=-N
     # - -dynamic
     dependencies:
-    - wallet-proxy
+    - mtl
     - optparse-applicative
+    - wallet-proxy
+    - warp
```
Comment (abizjak) [inline package.yaml:83]: Why is `warp` needed here explicitly? I don't see any changes in imports in the binary related to this.

**Review by abizjak — COMMENTED** [review]
Did you make sure that the https://jenkins.internal.concordium.com/job/wallet-proxy/ builds? 

I don't see any recent builds there.

File: package.yaml (hunk 1 above)
Comment (td202) [reply package.yaml:83]: I moved it from the general dependencies to here, because it is a dependency for the executable, but not for the library.

**Comment by td202** [comment]
> Did you make sure that the https://jenkins.internal.concordium.com/job/wallet-proxy/ builds?
> 
> I don't see any recent builds there.

Yes.

**Review by abizjak — APPROVED** [review]

**Pull request merged**